### ISSUE TITLE
Folders

### ISSUE THREAD

**Issue by TacticalLaptopBag** [markusfisch/PieLauncher#63]
I think it'd be really cool if the pie menu could have folders. I'm really digging the idea of having exactly 8 icons (and thus 8 directions) to swipe as it makes it super easy to get to apps without looking at my phone. However, I use more than 8 apps frequently, and adding more apps to the pie menu decreases the activation zones.

Solution: Folders. Just drag two icons on top of each other to create a folder. I have two ideas on how folder selection works.

Idea #1: when user swipes toward folder without releasing, the folder opens and the pie menu becomes the contents of the folder. Releasing without selecting an app closes the folder and the pie menu goes back to root. This could make folder interactions really snappy, but could result in some icons being outside the screen depending on where the user first taps.

Idea #2: the user selects the folder like any other app. When a folder is opened, the next time the user brings up the pie menu, it will be the folder contents. Tapping empty space will go up one folder. This solves the problem from the first option where apps could go outside the screen, tho it may be slightly more clunky. In this case, maybe there should be some sort of indicator to show that the pie menu is currently inside a folder.

**Comment by TacticalLaptopBag** [comment]
I just looked down the issues list and just saw this could be a duplicate of #52 😅 

Nonetheless, this would be really cool to have

**Comment by xi7** [comment]
easy idea is folder grid.view, really slow search my list games and unorganized

**Comment by TacticalLaptopBag** [comment]
Maybe the navigation could reset to root whenever the home screen focus is lost? AKA user goes into app drawer, opens app, locks screen, etc

I can see your point with losing muscle memory, but if the user always expects their pie menu to be at root when they first swap, they can memorize folder locations, and then locations within those folders. It might be complex, but if the user finds it confusing, they can simply not make any folders. Maybe having a different vibration pattern when selecting a folder could help.

Idk, I keep thinking about making a fork and giving it a go myself, but I have some other projects I want to get done first... maybe if this hasn't been touched by the time I'm done, I could give it a go

**Comment by xi7** [comment]
maybe good solution folders.

tunnel memory --- add long onSwipe(Left-Right-Top-Bottom) showing next muscle apps working like mini folders.

**Comment by xi7** [comment]
maybe good solution folders.

tunnel memory --- add long onSwipe(Left-Right-Top-Bottom) switching next muscle apps working like 4 mini folders.

**Comment by TacticalLaptopBag** [comment]
Ooh, so like idea 1, but the folders can only be in the main four directions? That could flow really well...

**Comment by xi7** [comment]
> Ooh, so like idea 1, but the folders can only be in the main four directions? That could flow really well...

thinking better.... we can use long swipe tunnel:
on bottom ( show notification )
on top ( lock phone )
left/right (actions)

about folders:
grid view folders to pie circle.
-after select folder will show apps.

**Comment by xi7** [comment]
> Whichever way this is implemented, it hampers the muscle memory btw. You supposed to swipe a specific direction & launch a specific app. Not any other actions. That's the only way I see it.
> 
> > I honestly donno why some need more than 7 to 8 apps simultaneously & frequently. A smartphone is just a tool. Put the top 8 apps which you use frequently & launch the rest from the drawer. That's not at all a problem, as the launcher already has auto launch perfect matching (where gboard's gesture swiping also negates typing too).

you are wrong, nobody will use launcher with unorganized gridview... i have 138 apps, search need memorize names keeping almost unusable without folder.

both can be implemented, tunnel and folder... even hide apps and overview.

**Comment by xi7** [comment]
> @xi7 If it's folders for drawer that you want, it can be done actually. But for pie home, it's not possible. The best layout for you, is grid (which is how all launchers work).

Basically a folder is an icon that could be added to home circle, like shortcut showing folder directly. why not ?

**Comment by xi7** [comment]
> That's what I'm referring to when I say you lose muscle memory. You have to swipe multiple times in multiple directions which turns out to be very complex to manage shortcuts.
> 
> Remember that a shortcut is always supposed to be a shortcut. Whatever changes this, will never be called a shortcut.


https://github.com/markusfisch/PieLauncher/assets/67331785/5ceacbeb-fee2-4097-98c0-707d8d065573

and now ?

**Comment by TacticalLaptopBag** [comment]
Ooh, I could see that style of folder being really useful in the app drawer, but for the pie menu itself, I was thinking it would just open another pie menu.

> I honestly donno why some need more than 7 to 8 apps simultaneously & frequently. A smartphone is just a tool. Put the top 8 apps which you use frequently & launch the rest from the drawer. That's not at all a problem, as the launcher already has auto launch perfect matching (where gboard's gesture swiping also negates typing too).

Some people have a lot of apps they use frequently, or use their phone for a myriad of things. My phone is basically my portable electronic powertool/swiss army knife. I have a lot of utility apps I use frequently, alongside the usual messaging, music, GPS, etc. Currently I'm getting around this with the Samsung app sidebar, but I'm not a huge fan of it while I have the much more satisfying pie menu. I'd just need one folder to fit all these apps in.

While I could just shove these apps onto my pie menu, it shrinks the swipe targets to be really small, and when I'm opening an app from the pie menu, I shouldn't have to be so precise with my gestures.

I think this is definitely mediated a lot by the ability to auto-open your keyboard when the app drawer opens, I have to bring my other hand to my phone to type the name out. With the pie menu, I could just swipe once, maybe twice depending on what I'm opening. But I'd still have the muscle memory of where it is.

Maybe this feature isn't suitable to all users, and in which case they can simply not use it. If contributors feel this feature doesn't belong here, then maybe a fork is in order
